### PR TITLE
Find all utters used in flows for validation

### PR DESCRIPTION
**Pull request by ancalita**
**Proposed changes**:
- Completes the remaining TODO-es listed in this [comment](https://rasahq.atlassian.net/browse/ATO-1389?focusedCommentId=20481).

**Status (please check what you already did)**:
- [X] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [X] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/validator.py [hunk 1]
```diff
@@ -249,8 +252,6 @@ def verify_utterances_in_dialogues(self, ignore_warnings: bool = True) -> bool:
         Checks whether utterances used in the stories are valid,
         and whether all valid utterances are used in stories.
         """
-        everything_is_alright = True
```
Comment (Urkem) [inline rasa/validator.py:252]: Why was this removed?

**Review by Urkem — APPROVED** [review]
LGTM (just one clarification)

File: rasa/validator.py (hunk 1 above)
Comment (ancalita) [reply rasa/validator.py:252]: My IDE signalled that it wasn't being in use since it's overwritten in L261.
